### PR TITLE
Refactor file structure

### PR DESCRIPTION
**Pull request by simiaosimis**
New File structure
Handle diagram click collision
Add gradient

### CODE REVIEW

File: src/jammed.js [hunk 1]
```diff
@@ -401,6 +385,141 @@ Method.prototype.draw = function(x, y) {
   }
   text += ")";
   apo.ctx.fillText(text, x+10+len.width, y);
+};console.log('loaded samplejs');
```
Comment (GustJc) [inline src/jammed.js]: This should not be "jammed" into jammed.js, but instead loaded into the html after jammed.js is loaded.

These functions bellow until line 433 are the .js the html will import after apoena to instanciate its canvas and elements.

File: src/jammed.js [hunk 2]
```diff
@@ -401,6 +385,141 @@ Method.prototype.draw = function(x, y) {
...
+  var canvas = document.getElementById('samplecanvas');
+  if (!canvas.getContext) {
+    console.log("Canvas not supported");
+    return;
+  }
+  var ctx = canvas.getContext('2d');
+  lines(ctx);
+  text(ctx);
+//
+//  ctx.fillRect(25,25,100,100);
```
Comment (GustJc) [inline src/jammed.js]: I know these comments aren't from the pull request, but they can be removed from src/test.js.
Just checked.... test.js is not used anymore, you can remove it from src/test.js

File: src/jam.c [hunk 3]
```diff
@@ -0,0 +1,144 @@
...
+
+
+#define file0 "core.js"
+#define file1 "line.js"
+#define file2 "inheritance_line.js"
+#define file3 "diagram.js"
+#define file4 "property_diagram.js"
+#define file5 "variable_property.js"
+#define file6 "method_property.js"
+#define file7 "test.js"
```
Comment (GustJc) [inline src/jam.c]: test.js should not be packed, it should be loaded in the html, as it instantiate the blocks for that specific html.

Just checked.... test.js is not used anymore, you can remove it from src/test.js

File: apoena.html [hunk 4]
```diff
@@ -3,26 +3,21 @@
...
 
-  <script type='text/javascript' src='../src/apoena.js'></script>
-  <script type='text/javascript' src='../src/apoena_parser.js'></script>
-  <script type='text/javascript' src='../src/sample.js'></script>
-  <style type="text/css">
-    canvas { border: 1px solid black; }
-    .todo { border: 1px solid black; background: rgb(230,230,230); width: 800px; padding-left: 10px; }
-    h2 { margin-bottom: 0px; }
-  </style>
+  <script type='text/javascript' src='src/sample.js'></script>
```
Comment (GustJc) [inline apoena.html:7]: For safety and better understanding, sample.js should be loaded after jammed.js, as it uses its functions and methods.

**Comment by simiaosimis** [comment]
@GustJc I removed the test.js file and changed the script so it doesn't look for this file

**Comment by GustJc** [comment]
@simiaosimis, you changed the name jammed.js to combined.js but forgot to update the qunit.html file. I'll merge it and fix this in a separate commit to be faster.

**Pull request merged**